### PR TITLE
Fb banner for internet

### PR DESCRIPTION
**Pull request by dingepfl**
This PR addresses the offline use of the app and when the internet connection is lost during the use of it (issue https://github.com/SwengGolfTeam/Sweng_Golf/issues/168).
A banner is displayed on all pages and the Firebase requests/queries are all cancelled.

The bug caused by the inactive location service https://github.com/SwengGolfTeam/Sweng_Golf/issues/172 is also addressed in the commits https://github.com/SwengGolfTeam/Sweng_Golf/commit/3b2e6970b4991b855196e97a4d259da927e94855 and https://github.com/SwengGolfTeam/Sweng_Golf/commit/1272eaa11d6661cff298b460c340bcc8a58fced5

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public void signInActivityCheckDialog(){
+        Config.quitTest();
+        Intents.init();
+        signInRule.launchActivity(new Intent());
+        Network.setFalseforTest(true);
+        Config.goToTest();
+        onView(withId(R.id.sign_in_button)).perform(click());
+        onView(withText(android.R.string.yes)).perform(click());
+        intended(toPackage("com.android.settings"));
+        //pressBack();
```
Comment (robinmamie) [inline app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java]: Don't forget to remove this line.

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void listOfferCheckDialog(){
+        Intents.init();
+        Network.setFalseforTest(true);
+        listOfferRule.launchActivity(new Intent());
+        onView(withText(android.R.string.yes)).perform(click());
+        intended(toPackage("com.android.settings"));
+        //pressBack();
```
Comment (robinmamie) [inline app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java]: Don't forget to remove this line.

File: app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java [hunk 3]
```diff
@@ -82,27 +85,30 @@ public void onComplete(DatabaseError databaseError,
...
     @Override
-    public void write(@NonNull String path, @NonNull String id, @NonNull Object object) {
+    public void write(@NonNull String path, @NonNull String id, @NonNull Object object){
         database.getReference(path).child(id).setValue(object);
     }
 
     @Override
     public void write(@NonNull String path, @NonNull String id, @NonNull Object object,
-                      @NonNull final CompletionListener listener) {
+    @NonNull final CompletionListener listener){
```
Comment (robinmamie) [inline app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java]: Did you reformat the code propoerly? The spacing seems odd. Maybe you could reformat the whole java folder to be sure that everything looks good.

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java (hunk 1 above)
Comment (dingepfl) [reply app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java]: I have removed it in the upcoming commit, thank you!

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java (hunk 2 above)
Comment (dingepfl) [reply app/src/androidTest/java/ch/epfl/sweng/swenggolf/OfflineTests.java]: I have removed it in the upcoming commit, thank you!

File: app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java (hunk 3 above)
Comment (dingepfl) [reply app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java]: I think I did but I will redo it anyway to make sure, good catch

**Comment by dingepfl** [comment]
See PR https://github.com/SwengGolfTeam/Sweng_Golf/pull/182 for the actual merge to master...

**Pull request closed without merging**